src/components/JSONeditor/JSONeditor.js

```diff
@@ -7,22 +12,35 @@ import 'brace/ext/searchbox'
 
 import './JSONeditor.css'
 
-const JSONeditor = props => (
-  <AceEditor
-    mode='json'
-    theme='github'
-    className='json-editor'
-    name='gw-json-editor'
-    readOnly={props.readOnly}
-    onChange={props.onChange}
-    value={props.value}
-    editorProps={{
-      highlightActiveLine: true,
-      showGutter: true,
-      showLineNumbers: true,
-      $blockScrolling: Infinity
-    }}
-  />
-)
+const propTypes = {
+  value: string.isRequired,
+  isShown: bool.isRequired,
+  onChange: func,
+  readOnly: bool
+}
+
+const JSONeditor = props => {
+  if (!props.isShown) return null
```

Comment:
just `return` would be enough I guess

Reply:
:)
I would agree with you, but `React` doesn't :) It require to return explicit `null` if you don't need to render anything :(